<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 6.3.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="Downstream RNA-Seq analysis utilizes R for differential gene expression analysis. I will show my work of salmon quantification data management with R in this post.Thanks to Bioconductor for providing">
<meta property="og:type" content="article">
<meta property="og:title" content="RNA-Seq Downstream Workflow: differential expression gene analysis with Salmon quantification">
<meta property="og:url" content="http://example.com/2023/02/25/RNA-Seq-downstream-analysis-differential-expression-gene-analysis-with-Salmon-quantification/index.html">
<meta property="og:site_name" content="BIOINFOISLAND">
<meta property="og:description" content="Downstream RNA-Seq analysis utilizes R for differential gene expression analysis. I will show my work of salmon quantification data management with R in this post.Thanks to Bioconductor for providing">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2023-02-25T06:43:33.000Z">
<meta property="article:modified_time" content="2023-03-01T14:05:18.690Z">
<meta property="article:author" content="Qiayi">
<meta property="article:tag" content="RNA-Seq">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="http://example.com/2023/02/25/RNA-Seq-downstream-analysis-differential-expression-gene-analysis-with-Salmon-quantification/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>RNA-Seq Downstream Workflow: differential expression gene analysis with Salmon quantification | BIOINFOISLAND</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">BIOINFOISLAND</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2023/02/25/RNA-Seq-downstream-analysis-differential-expression-gene-analysis-with-Salmon-quantification/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="http://img.name2012.com/uploads/allimg/2015-04/26-011155_74.jpg">
      <meta itemprop="name" content="Qiayi">
      <meta itemprop="description" content="An island where we record bioinformatics learning process.">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="BIOINFOISLAND">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          RNA-Seq Downstream Workflow: differential expression gene analysis with Salmon quantification
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2023-02-25 14:43:33" itemprop="dateCreated datePublished" datetime="2023-02-25T14:43:33+08:00">2023-02-25</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2023-03-01 22:05:18" itemprop="dateModified" datetime="2023-03-01T22:05:18+08:00">2023-03-01</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/RNA-Seq-Workflow/" itemprop="url" rel="index"><span itemprop="name">RNA-Seq Workflow</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <p>Downstream RNA-Seq analysis utilizes R for differential gene expression analysis. I will show my work of salmon quantification data management with R in this post.<br>Thanks to <a target="_blank" rel="noopener" href="http://bioconductor.org/packages/release/workflows/vignettes/rnaseqDTU/inst/doc/rnaseqDTU.html#dge-analysis-with-deseq2">Bioconductor</a> for providing helpful vignettes about RNA-Seq downstream analysis and visualization.</p>
<span id="more"></span>

<h3 id="Outline"><a href="#Outline" class="headerlink" title="Outline:"></a>Outline:</h3><ul>
<li>Import the quant.sf files</li>
<li>Obtain read counts</li>
<li>PCA: Check the experimental reliability</li>
<li>Transfer Ensembl id to gene id </li>
<li>Analyze the differential gene expression with DESeq2</li>
</ul>
<h3 id="Import-the-quant-sf-files"><a href="#Import-the-quant-sf-files" class="headerlink" title="Import the quant.sf files"></a>Import the quant.sf files</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span><span class="string">&quot;tximport&quot;</span><span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span><span class="string">&quot;readr&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#file path</span></span><br><span class="line">setwd<span class="punctuation">(</span><span class="string">&#x27;D:/yumiiiii/Feb_info&#x27;</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment">#check the path</span></span><br><span class="line">getwd<span class="punctuation">(</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment">#Please compile all the quant.sf file in a package called salmon.output</span></span><br><span class="line">dir <span class="operator">&lt;-</span> <span class="string">&#x27;D:/yumiiiii/Feb_info/salmon.output&#x27;</span></span><br><span class="line"><span class="comment">#cat the quant.sf file path</span></span><br><span class="line">files <span class="operator">&lt;-</span> file.path<span class="punctuation">(</span>dir<span class="punctuation">,</span> list.files<span class="punctuation">(</span><span class="string">&quot;salmon.output&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="string">&quot;quant.sf&quot;</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
<p>The file structure is as follow.</p>
<p><strong>Output for <code>files</code>:</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="punctuation">[</span><span class="number">1</span><span class="punctuation">]</span> <span class="string">&quot;D:/yumiiiii/Feb_info/salmon.output/468-sgM1-1_1/quant.sf&quot;</span></span><br><span class="line"><span class="punctuation">[</span><span class="number">2</span><span class="punctuation">]</span> <span class="string">&quot;D:/yumiiiii/Feb_info/salmon.output/468-sgM1-2_1/quant.sf&quot;</span></span><br><span class="line"><span class="punctuation">[</span><span class="number">3</span><span class="punctuation">]</span> <span class="string">&quot;D:/yumiiiii/Feb_info/salmon.output/468-sgM3-1_1/quant.sf&quot;</span></span><br><span class="line"><span class="punctuation">[</span><span class="number">4</span><span class="punctuation">]</span> <span class="string">&quot;D:/yumiiiii/Feb_info/salmon.output/468-sgM3-2_1/quant.sf&quot;</span></span><br><span class="line"><span class="punctuation">[</span><span class="number">5</span><span class="punctuation">]</span> <span class="string">&quot;D:/yumiiiii/Feb_info/salmon.output/468-sgNT-1_1/quant.sf&quot;</span></span><br><span class="line"><span class="punctuation">[</span><span class="number">6</span><span class="punctuation">]</span> <span class="string">&quot;D:/yumiiiii/Feb_info/salmon.output/468-sgNT-2_1/quant.sf&quot;</span></span><br></pre></td></tr></table></figure>

<h3 id="Obtain-read-counts"><a href="#Obtain-read-counts" class="headerlink" title="Obtain read counts"></a>Obtain read counts</h3><p>Import the reference genome. Here, we use <code>gtf</code> files on <code>gencode</code> website. This latest version of this file is reliable. </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span><span class="string">&#x27;GenomicFeatures&#x27;</span><span class="punctuation">)</span></span><br><span class="line">gtf <span class="operator">&lt;-</span> <span class="string">&quot;gencode.v43.chr_patch_hapl_scaff.annotation.gtf.gz&quot;</span></span><br><span class="line">txdb.filename <span class="operator">&lt;-</span> <span class="string">&quot;gencode.v43.chr_patch_hapl_scaff.annotation.gtf.sqlite&quot;</span></span><br><span class="line">txdb <span class="operator">&lt;-</span> makeTxDbFromGFF<span class="punctuation">(</span>gtf<span class="punctuation">)</span></span><br><span class="line">saveDb<span class="punctuation">(</span>txdb<span class="punctuation">,</span> txdb.filename<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">## Once the TxDb database has been generated and saved, it can be quickly reloaded:</span></span><br><span class="line">txdb <span class="operator">&lt;-</span> loadDb<span class="punctuation">(</span>txdb.filename<span class="punctuation">)</span></span><br><span class="line">txdf <span class="operator">&lt;-</span> select<span class="punctuation">(</span>txdb<span class="punctuation">,</span> keys<span class="punctuation">(</span>txdb<span class="punctuation">,</span> <span class="string">&quot;GENEID&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span> <span class="string">&quot;TXNAME&quot;</span><span class="punctuation">,</span> <span class="string">&quot;GENEID&quot;</span><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># generate the matrix of counts from quantification files</span></span><br><span class="line"><span class="comment"># requires tx2gene for gene-level summarization</span></span><br><span class="line">txi.g <span class="operator">&lt;-</span> tximport<span class="punctuation">(</span>files<span class="punctuation">,</span> type <span class="operator">=</span> <span class="string">&quot;salmon&quot;</span><span class="punctuation">,</span> tx2gene <span class="operator">=</span> txdf<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">2</span><span class="operator">:</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>txi.g<span class="operator">$</span>counts<span class="punctuation">)</span> </span><br></pre></td></tr></table></figure>

<h3 id="PCA-Check-the-experimental-replica"><a href="#PCA-Check-the-experimental-replica" class="headerlink" title="PCA: Check the experimental replica"></a>PCA: Check the experimental replica</h3><p>The experiment is conduced twice. We have to check whether the replica is reasonable. We use PCA to cluster the 3 groups. If they are perfectly clustered, we could verify the reasonability of the experiment.  </p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br><span class="line">23</span><br><span class="line">24</span><br><span class="line">25</span><br><span class="line">26</span><br><span class="line">27</span><br><span class="line">28</span><br><span class="line">29</span><br><span class="line">30</span><br><span class="line">31</span><br><span class="line">32</span><br><span class="line">33</span><br><span class="line">34</span><br><span class="line">35</span><br><span class="line">36</span><br><span class="line">37</span><br><span class="line">38</span><br><span class="line">39</span><br><span class="line">40</span><br></pre></td><td class="code"><pre><span class="line">gene <span class="operator">&lt;-</span> as.matrix<span class="punctuation">(</span>txi.g<span class="operator">$</span>abundance<span class="punctuation">)</span> </span><br><span class="line">gene <span class="operator">&lt;-</span> t<span class="punctuation">(</span>gene<span class="punctuation">)</span></span><br><span class="line">library<span class="punctuation">(</span>FactoMineR<span class="punctuation">)</span></span><br><span class="line">gene.pca <span class="operator">&lt;-</span> PCA<span class="punctuation">(</span>gene<span class="punctuation">,</span> ncp <span class="operator">=</span> <span class="number">2</span><span class="punctuation">,</span> scale.unit <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">,</span> graph <span class="operator">=</span> <span class="literal">FALSE</span><span class="punctuation">)</span></span><br><span class="line">plot<span class="punctuation">(</span>gene.pca<span class="punctuation">)</span>  <span class="comment">#PCA single plot</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">#Beauty the plot with ggplot2</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment">#Extract PC1 &amp; PC2</span></span><br><span class="line">pca_sample <span class="operator">&lt;-</span> data.frame<span class="punctuation">(</span>gene.pca<span class="operator">$</span>ind<span class="operator">$</span>coord<span class="punctuation">[</span> <span class="punctuation">,</span><span class="number">1</span><span class="operator">:</span><span class="number">2</span><span class="punctuation">]</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>pca_sample<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#Extract the contribution of PC1 &amp; PC2</span></span><br><span class="line">pca_eig1 <span class="operator">&lt;-</span> <span class="built_in">round</span><span class="punctuation">(</span>gene.pca<span class="operator">$</span>eig<span class="punctuation">[</span><span class="number">1</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">]</span><span class="punctuation">,</span> <span class="number">2</span><span class="punctuation">)</span></span><br><span class="line">pca_eig2 <span class="operator">&lt;-</span> <span class="built_in">round</span><span class="punctuation">(</span>gene.pca<span class="operator">$</span>eig<span class="punctuation">[</span><span class="number">2</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">]</span><span class="punctuation">,</span><span class="number">2</span> <span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line"></span><br><span class="line"><span class="comment"># read and merge the sample information </span></span><br><span class="line">group <span class="operator">=</span> data.frame<span class="punctuation">(</span></span><br><span class="line">  condition <span class="operator">=</span> <span class="built_in">rep</span><span class="punctuation">(</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;KO1&quot;</span><span class="punctuation">,</span><span class="string">&quot;KO3&quot;</span><span class="punctuation">,</span><span class="string">&quot;NT&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span>each <span class="operator">=</span> <span class="number">2</span><span class="punctuation">)</span></span><br><span class="line"><span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">group <span class="operator">&lt;-</span> group<span class="punctuation">[</span>rownames<span class="punctuation">(</span>pca_sample<span class="punctuation">)</span><span class="punctuation">,</span> <span class="punctuation">]</span></span><br><span class="line">pca_sample <span class="operator">&lt;-</span> cbind<span class="punctuation">(</span>pca_sample<span class="punctuation">,</span> group<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">pca_sample<span class="operator">$</span>samples <span class="operator">&lt;-</span> rownames<span class="punctuation">(</span>pca_sample<span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>pca_sample<span class="punctuation">)</span> </span><br><span class="line"></span><br><span class="line"><span class="comment"># ggplot2: draw the dot plot</span></span><br><span class="line">library<span class="punctuation">(</span>ggplot2<span class="punctuation">)</span></span><br><span class="line"></span><br><span class="line">p <span class="operator">&lt;-</span> ggplot<span class="punctuation">(</span>data <span class="operator">=</span> pca_sample<span class="punctuation">,</span> aes<span class="punctuation">(</span>x <span class="operator">=</span> pca_sample<span class="operator">$</span>Dim.1<span class="punctuation">,</span> y <span class="operator">=</span> pca_sample<span class="operator">$</span>Dim.2<span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">+</span></span><br><span class="line">  geom_point<span class="punctuation">(</span>aes<span class="punctuation">(</span>color <span class="operator">=</span> group<span class="punctuation">)</span><span class="punctuation">,</span> size <span class="operator">=</span> <span class="number">5</span><span class="punctuation">)</span> <span class="operator">+</span> </span><br><span class="line">  scale_color_manual<span class="punctuation">(</span>values <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&#x27;orange&#x27;</span><span class="punctuation">,</span> <span class="string">&#x27;purple&#x27;</span><span class="punctuation">,</span><span class="string">&#x27;green&#x27;</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">+</span> </span><br><span class="line">  theme<span class="punctuation">(</span>panel.grid <span class="operator">=</span> element_blank<span class="punctuation">(</span><span class="punctuation">)</span><span class="punctuation">,</span> panel.background <span class="operator">=</span> element_rect<span class="punctuation">(</span>color <span class="operator">=</span> <span class="string">&#x27;black&#x27;</span><span class="punctuation">,</span> fill <span class="operator">=</span> <span class="string">&#x27;transparent&#x27;</span><span class="punctuation">)</span><span class="punctuation">,</span> </span><br><span class="line">        legend.key <span class="operator">=</span> element_rect<span class="punctuation">(</span>fill <span class="operator">=</span> <span class="string">&#x27;transparent&#x27;</span><span class="punctuation">)</span><span class="punctuation">)</span> <span class="operator">+</span>  </span><br><span class="line">  labs<span class="punctuation">(</span>x <span class="operator">=</span>  paste<span class="punctuation">(</span><span class="string">&#x27;PCA1:&#x27;</span><span class="punctuation">,</span> pca_eig1<span class="punctuation">,</span> <span class="string">&#x27;%&#x27;</span><span class="punctuation">)</span><span class="punctuation">,</span> y <span class="operator">=</span> paste<span class="punctuation">(</span><span class="string">&#x27;PCA2:&#x27;</span><span class="punctuation">,</span> pca_eig2<span class="punctuation">,</span> <span class="string">&#x27;%&#x27;</span><span class="punctuation">)</span><span class="punctuation">,</span> color <span class="operator">=</span> <span class="string">&#x27;&#x27;</span><span class="punctuation">)</span>  </span><br><span class="line">p</span><br></pre></td></tr></table></figure>

<h3 id="Transfer-Ensembl-id-to-gene-id"><a href="#Transfer-Ensembl-id-to-gene-id" class="headerlink" title="Transfer Ensembl id to gene id"></a>Transfer Ensembl id to gene id</h3><p>Step1. we have to remove the decimals of Ensembl id before converting them to gene id.</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">cts <span class="operator">&lt;-</span> data.frame<span class="punctuation">(</span>txi.g<span class="operator">$</span>counts<span class="punctuation">)</span></span><br><span class="line">rownames<span class="punctuation">(</span>cts<span class="punctuation">)</span> <span class="operator">&lt;-</span> gsub<span class="punctuation">(</span><span class="string">&#x27;\\..*&#x27;</span><span class="punctuation">,</span><span class="string">&#x27;&#x27;</span><span class="punctuation">,</span>rownames<span class="punctuation">(</span>cts<span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">cts<span class="operator">$</span>id <span class="operator">&lt;-</span> rownames<span class="punctuation">(</span>cts<span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>cts<span class="punctuation">)</span></span><br></pre></td></tr></table></figure>

<p><strong>Output after removing decimals:</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator">&gt;</span> head<span class="punctuation">(</span>countData<span class="punctuation">)</span></span><br><span class="line">                X468.sgM1.1_1 X468.sgM1.2_1 X468.sgNT.1_1 X468.sgNT.2_1</span><br><span class="line">ENSG00000000003        <span class="number">58.722</span>        <span class="number">58.142</span>        <span class="number">20.065</span>        <span class="number">35.244</span></span><br><span class="line">ENSG00000000005         <span class="number">0.000</span>         <span class="number">0.000</span>         <span class="number">0.000</span>         <span class="number">0.000</span></span><br><span class="line">ENSG00000000419        <span class="number">74.293</span>        <span class="number">48.119</span>       <span class="number">102.646</span>        <span class="number">40.919</span></span><br><span class="line">ENSG00000000457        <span class="number">78.161</span>        <span class="number">66.016</span>        <span class="number">86.334</span>        <span class="number">63.542</span></span><br><span class="line">ENSG00000000460       <span class="number">135.999</span>        <span class="number">94.274</span>       <span class="number">209.964</span>       <span class="number">161.174</span></span><br><span class="line">ENSG00000000938         <span class="number">2.000</span>         <span class="number">6.000</span>         <span class="number">0.000</span>         <span class="number">0.000</span></span><br></pre></td></tr></table></figure>


<p>Step2. We could transfer Ensembl id to gene id with a table obtained from gtf file mentioned before.</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">library<span class="punctuation">(</span><span class="string">&quot;rtracklayer&quot;</span><span class="punctuation">)</span></span><br><span class="line"><span class="comment">#gtf path</span></span><br><span class="line">gtf_data <span class="operator">=</span> import<span class="punctuation">(</span><span class="string">&#x27;gencode.v43.chr_patch_hapl_scaff.annotation.gtf&#x27;</span><span class="punctuation">)</span> </span><br><span class="line"></span><br><span class="line"><span class="comment">#import gtf file to form a GRangs object</span></span><br><span class="line">gtf_data <span class="operator">=</span> as.data.frame<span class="punctuation">(</span>gtf_data<span class="punctuation">)</span></span><br><span class="line">gtf_data <span class="operator">&lt;-</span> data.frame<span class="punctuation">(</span>gtf_data<span class="operator">$</span>gene_id<span class="punctuation">,</span>gtf_data<span class="operator">$</span>gene_name<span class="punctuation">)</span></span><br><span class="line">colnames<span class="punctuation">(</span>gtf_data<span class="punctuation">)</span> <span class="operator">&lt;-</span>  <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&#x27;ensembl_id&#x27;</span><span class="punctuation">,</span><span class="string">&#x27;gene_id&#x27;</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>gtf_data<span class="punctuation">)</span></span><br><span class="line">table<span class="punctuation">(</span>duplicated<span class="punctuation">(</span>gtf_data<span class="punctuation">[</span><span class="punctuation">,</span><span class="number">1</span><span class="punctuation">]</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
<p><strong>Output for transfer table <code>ids</code></strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator">&gt;</span> head<span class="punctuation">(</span>gtf_data<span class="punctuation">)</span></span><br><span class="line"> </span><br><span class="line">ensembl_id<span class="operator">&lt;</span>chr<span class="operator">&gt;</span>   gene_id<span class="operator">&lt;</span>chr<span class="operator">&gt;</span></span><br><span class="line"><span class="number">1</span>	ENSG00000290825	DDX11L2		</span><br><span class="line"><span class="number">6</span>	ENSG00000223972	DDX11L1		</span><br><span class="line"><span class="number">14</span>	ENSG00000227232	WASH7P		</span><br><span class="line"><span class="number">27</span>	ENSG00000278267	MIR6859<span class="operator">-</span><span class="number">1</span>		</span><br><span class="line"><span class="number">30</span>	ENSG00000243485	MIR1302<span class="operator">-</span><span class="number">2</span>HG		</span><br><span class="line"><span class="number">38</span>	ENSG00000284332	MIR1302<span class="operator">-</span><span class="number">2</span></span><br></pre></td></tr></table></figure>
<p>Step3. Join two tables and remove the duplicate gene id items.</p>
<figure class="highlight plaintext"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">library(&#x27;dplyr&#x27;)</span><br><span class="line">index &lt;- duplicated(gtf_data[,2])</span><br><span class="line">gtf_data &lt;- gtf_data[!index,]</span><br><span class="line">gtf_data$ensembl_id &lt;- gsub(&#x27;\\..*&#x27;,&#x27;&#x27;,gtf_data$ensembl_id)</span><br><span class="line">table(duplicated(gtf_data[,1]))</span><br><span class="line">df=left_join(cts,gtf_data,by=c(&quot;id&quot;=&quot;ensembl_id&quot;)) </span><br><span class="line">table(duplicated(df$gene_id))</span><br><span class="line">head(df)</span><br></pre></td></tr></table></figure>
<p><strong>Output after conversion:</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator">&gt;</span> head<span class="punctuation">(</span>conv2<span class="punctuation">)</span></span><br><span class="line">         X468.sgM1.1_1 X468.sgM1.2_1 X468.sgNT.1_1 X468.sgNT.2_1</span><br><span class="line">TSPAN6          <span class="number">58.722</span>        <span class="number">58.142</span>        <span class="number">20.065</span>        <span class="number">35.244</span></span><br><span class="line">TNMD             <span class="number">0.000</span>         <span class="number">0.000</span>         <span class="number">0.000</span>         <span class="number">0.000</span></span><br><span class="line">DPM1            <span class="number">74.293</span>        <span class="number">48.119</span>       <span class="number">102.646</span>        <span class="number">40.919</span></span><br><span class="line">SCYL3           <span class="number">78.161</span>        <span class="number">66.016</span>        <span class="number">86.334</span>        <span class="number">63.542</span></span><br><span class="line">C1orf112       <span class="number">135.999</span>        <span class="number">94.274</span>       <span class="number">209.964</span>       <span class="number">161.174</span></span><br><span class="line">FGR              <span class="number">2.000</span>         <span class="number">6.000</span>         <span class="number">0.000</span>         <span class="number">0.000</span></span><br></pre></td></tr></table></figure>

<h3 id="Analyze-the-differential-gene-expression-with-DESeq2"><a href="#Analyze-the-differential-gene-expression-with-DESeq2" class="headerlink" title="Analyze the differential gene expression with DESeq2"></a>Analyze the differential gene expression with DESeq2</h3><figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#metadata</span></span><br><span class="line">coldata <span class="operator">=</span> data.frame<span class="punctuation">(</span></span><br><span class="line">  condition <span class="operator">=</span> <span class="built_in">rep</span><span class="punctuation">(</span><span class="built_in">c</span><span class="punctuation">(</span><span class="string">&quot;KO&quot;</span><span class="punctuation">,</span><span class="string">&quot;NT&quot;</span><span class="punctuation">)</span><span class="punctuation">,</span>time <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="number">4</span><span class="punctuation">,</span><span class="number">2</span><span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line"><span class="punctuation">)</span></span><br><span class="line">coldata</span><br><span class="line"></span><br><span class="line"><span class="comment">### remove replicated ids and NA</span></span><br><span class="line">index1 <span class="operator">&lt;-</span> duplicated<span class="punctuation">(</span>df<span class="operator">$</span>gene_id<span class="punctuation">)</span></span><br><span class="line">table<span class="punctuation">(</span>index1<span class="punctuation">)</span></span><br><span class="line">df <span class="operator">&lt;-</span> df<span class="punctuation">[</span><span class="operator">!</span>duplicated<span class="punctuation">(</span>df<span class="operator">$</span>gene_id<span class="punctuation">)</span><span class="punctuation">,</span><span class="punctuation">]</span></span><br><span class="line">table<span class="punctuation">(</span>duplicated<span class="punctuation">(</span>df<span class="operator">$</span>gene_id<span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">df <span class="operator">&lt;-</span> na.omit<span class="punctuation">(</span>df<span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>df<span class="punctuation">)</span></span><br><span class="line"><span class="built_in">dim</span><span class="punctuation">(</span>df<span class="punctuation">)</span></span><br><span class="line">rownames<span class="punctuation">(</span>df<span class="punctuation">)</span> <span class="operator">&lt;-</span> df<span class="operator">$</span>gene_id</span><br></pre></td></tr></table></figure>

<p><strong>Output for coldata:</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator">&gt;</span> coldata</span><br><span class="line"> condition <span class="operator">&lt;</span>chr<span class="operator">&gt;</span></span><br><span class="line">KO				</span><br><span class="line">KO				</span><br><span class="line">KO				</span><br><span class="line">KO				</span><br><span class="line">NT				</span><br><span class="line">NT</span><br></pre></td></tr></table></figure>

<p>At last, we could use DESeq2 package for differential expression analysis.</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#TRUE</span></span><br><span class="line">library<span class="punctuation">(</span><span class="string">&quot;DESeq2&quot;</span><span class="punctuation">)</span></span><br><span class="line">dds <span class="operator">&lt;-</span> DESeqDataSetFromMatrix<span class="punctuation">(</span>countData <span class="operator">=</span> <span class="built_in">round</span><span class="punctuation">(</span>cts<span class="punctuation">)</span><span class="punctuation">,</span></span><br><span class="line">                              colData <span class="operator">=</span> coldata<span class="punctuation">,</span></span><br><span class="line">                              design <span class="operator">=</span> <span class="operator">~</span> condition<span class="punctuation">)</span></span><br><span class="line">dds</span><br><span class="line">dds <span class="operator">&lt;-</span> DESeq<span class="punctuation">(</span>dds<span class="punctuation">)</span></span><br><span class="line"><span class="comment">#set contrast</span></span><br><span class="line">b <span class="operator">&lt;-</span> <span class="string">&#x27;KO&#x27;</span></span><br><span class="line">a <span class="operator">&lt;-</span> <span class="string">&#x27;NT&#x27;</span></span><br><span class="line"></span><br><span class="line">result <span class="operator">&lt;-</span> results<span class="punctuation">(</span>dds<span class="punctuation">,</span>contrast <span class="operator">=</span> <span class="built_in">c</span><span class="punctuation">(</span><span class="string">&#x27;condition&#x27;</span><span class="punctuation">,</span>b<span class="punctuation">,</span>a<span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>result<span class="punctuation">)</span></span><br><span class="line">head<span class="punctuation">(</span>dds<span class="punctuation">)</span></span><br><span class="line"><span class="comment">#save the result for DESeq2</span></span><br><span class="line">write.csv<span class="punctuation">(</span>result<span class="punctuation">,</span><span class="string">&#x27;./output/KO_VS_NT.csv&#x27;</span><span class="punctuation">,</span>row.names <span class="operator">=</span> <span class="literal">TRUE</span><span class="punctuation">)</span></span><br></pre></td></tr></table></figure>
<p><strong>Output for DESeq2 report:</strong></p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="operator">&gt;</span> head<span class="punctuation">(</span>results<span class="punctuation">(</span>dds<span class="punctuation">)</span><span class="punctuation">)</span></span><br><span class="line">log2 fold change <span class="punctuation">(</span>MLE<span class="punctuation">)</span><span class="operator">:</span> condition treated vs control </span><br><span class="line">Wald test p<span class="operator">-</span>value<span class="operator">:</span> condition treated vs control </span><br><span class="line">DataFrame with <span class="number">6</span> rows and <span class="number">6</span> columns</span><br><span class="line">          baseMean log2FoldChange     lfcSE      stat    pvalue      padj    </span><br><span class="line">         <span class="operator">&lt;</span>numeric<span class="operator">&gt;</span>      <span class="operator">&lt;</span>numeric<span class="operator">&gt;</span> <span class="operator">&lt;</span>numeric<span class="operator">&gt;</span> <span class="operator">&lt;</span>numeric<span class="operator">&gt;</span> <span class="operator">&lt;</span>numeric<span class="operator">&gt;</span> <span class="operator">&lt;</span>numeric<span class="operator">&gt;</span></span><br><span class="line">TSPAN6    <span class="number">42.50922</span>      <span class="number">1.0717692</span>  <span class="number">0.704616</span>  <span class="number">1.521068</span>  <span class="number">0.128243</span>  <span class="number">0.423952</span></span><br><span class="line">TNMD       <span class="number">0.00000</span>             <span class="literal">NA</span>        <span class="literal">NA</span>        <span class="literal">NA</span>        <span class="literal">NA</span>        <span class="literal">NA</span></span><br><span class="line">DPM1      <span class="number">66.17747</span>     <span class="operator">-</span><span class="number">0.2424019</span>  <span class="number">0.715360</span> <span class="operator">-</span><span class="number">0.338853</span>  <span class="number">0.734720</span>  <span class="number">0.925294</span></span><br><span class="line">SCYL3     <span class="number">72.93360</span>     <span class="operator">-</span><span class="number">0.0710815</span>  <span class="number">0.533399</span> <span class="operator">-</span><span class="number">0.133261</span>  <span class="number">0.893987</span>  <span class="number">0.974060</span></span><br><span class="line">C1orf112 <span class="number">149.48464</span>     <span class="operator">-</span><span class="number">0.6957886</span>  <span class="number">0.435900</span> <span class="operator">-</span><span class="number">1.596210</span>  <span class="number">0.110442</span>  <span class="number">0.385793</span></span><br><span class="line">FGR        <span class="number">1.92224</span>      <span class="number">4.3921508</span>  <span class="number">3.821075</span>  <span class="number">1.149454</span>  <span class="number">0.250369</span>        <span class="literal">NA</span></span><br></pre></td></tr></table></figure>
    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/RNA-Seq/" rel="tag"># RNA-Seq</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2023/02/24/004-Linux%E5%AD%A6%E4%B9%A004-%E5%B8%88%E5%85%84%E8%AF%B4%EF%BC%9A%E7%94%A8%E4%B8%80%E4%B8%AA%E5%91%BD%E4%BB%A4%E8%BE%93%E5%87%BA%E4%B8%8A%E6%B8%B8%E5%88%86%E6%9E%90%E7%9A%84%E6%89%80%E6%9C%89%E7%BB%93%E6%9E%9C/" rel="prev" title="004- Linux学习04-师兄说：用一个命令输出RNA-Seq上游分析的所有结果">
      <i class="fa fa-chevron-left"></i> 004- Linux学习04-师兄说：用一个命令输出RNA-Seq上游分析的所有结果
    </a></div>
      <div class="post-nav-item">
    <a href="/2023/02/27/005-r%E8%AF%AD%E8%A8%80-%E5%AF%BC%E5%85%A5%E6%96%87%E4%BB%B6%E8%B7%AF%E5%BE%84/" rel="next" title="005- r语言-导入文件路径:`list.file`和`file.path`">
      005- r语言-导入文件路径:`list.file`和`file.path` <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#Outline"><span class="nav-number">1.</span> <span class="nav-text">Outline:</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Import-the-quant-sf-files"><span class="nav-number">2.</span> <span class="nav-text">Import the quant.sf files</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Obtain-read-counts"><span class="nav-number">3.</span> <span class="nav-text">Obtain read counts</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#PCA-Check-the-experimental-replica"><span class="nav-number">4.</span> <span class="nav-text">PCA: Check the experimental replica</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Transfer-Ensembl-id-to-gene-id"><span class="nav-number">5.</span> <span class="nav-text">Transfer Ensembl id to gene id</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Analyze-the-differential-gene-expression-with-DESeq2"><span class="nav-number">6.</span> <span class="nav-text">Analyze the differential gene expression with DESeq2</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Qiayi"
      src="http://img.name2012.com/uploads/allimg/2015-04/26-011155_74.jpg">
  <p class="site-author-name" itemprop="name">Qiayi</p>
  <div class="site-description" itemprop="description">An island where we record bioinformatics learning process.</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">9</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">7</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">6</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2023-02 – 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Qiayi</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
