<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 6.3.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="Overview FastQC (-&gt; Multiqc): Quality control Trim_galore: Trim adapters Mapping (Salmon&#x2F; HISAT2+StringTie)">
<meta property="og:type" content="article">
<meta property="og:title" content="RNA-Seq Upstream Workflow">
<meta property="og:url" content="http://example.com/2023/02/24/RNA-Seq-upstream-workflow/index.html">
<meta property="og:site_name" content="BIOINFOISLAND">
<meta property="og:description" content="Overview FastQC (-&gt; Multiqc): Quality control Trim_galore: Trim adapters Mapping (Salmon&#x2F; HISAT2+StringTie)">
<meta property="og:locale" content="en_US">
<meta property="og:image" content="http://example.com/RNA-Seq-upstream-workflow/fastqc_output.png">
<meta property="og:image" content="http://example.com/RNA-Seq-upstream-workflow/multiqc_output.png">
<meta property="article:published_time" content="2023-02-24T06:35:56.000Z">
<meta property="article:modified_time" content="2023-02-25T12:07:50.063Z">
<meta property="article:author" content="Qiayi">
<meta property="article:tag" content="RNA-Seq">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="http://example.com/RNA-Seq-upstream-workflow/fastqc_output.png">

<link rel="canonical" href="http://example.com/2023/02/24/RNA-Seq-upstream-workflow/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>RNA-Seq Upstream Workflow | BIOINFOISLAND</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">BIOINFOISLAND</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2023/02/24/RNA-Seq-upstream-workflow/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="http://img.name2012.com/uploads/allimg/2015-04/26-011155_74.jpg">
      <meta itemprop="name" content="Qiayi">
      <meta itemprop="description" content="An island where we record bioinformatics learning process.">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="BIOINFOISLAND">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          RNA-Seq Upstream Workflow
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2023-02-24 14:35:56" itemprop="dateCreated datePublished" datetime="2023-02-24T14:35:56+08:00">2023-02-24</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2023-02-25 20:07:50" itemprop="dateModified" datetime="2023-02-25T20:07:50+08:00">2023-02-25</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/RNA-Seq-Workflow/" itemprop="url" rel="index"><span itemprop="name">RNA-Seq Workflow</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h3 id="Overview"><a href="#Overview" class="headerlink" title="Overview"></a>Overview</h3><ul>
<li>FastQC (-&gt; Multiqc): Quality control</li>
<li>Trim_galore: Trim adapters</li>
<li>Mapping (Salmon&#x2F; HISAT2+StringTie)</li>
</ul>
<span id="more"></span>

<h3 id="FastQC-Multiqc"><a href="#FastQC-Multiqc" class="headerlink" title="FastQC (+ Multiqc)"></a>FastQC (+ Multiqc)</h3><h4 id="Install"><a href="#Install" class="headerlink" title="Install:"></a><strong>Install:</strong></h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">conda install fastqc -c bioconda</span><br><span class="line">conda install multiqc -c bioconda</span><br></pre></td></tr></table></figure>

<p>Most tools for bio can be installed via <strong>bioconda channel</strong>.</p>
<h4 id="Run-fastQC"><a href="#Run-fastQC" class="headerlink" title="Run fastQC:"></a><strong>Run fastQC:</strong></h4><p><strong>Input:</strong> .fq.gz file from raw data </p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># fastQC</span></span><br><span class="line"><span class="comment"># -o output dir </span></span><br><span class="line"><span class="comment"># -t thread</span></span><br><span class="line"><span class="comment"># input dir is at the end of the command</span></span><br><span class="line"></span><br><span class="line">fastqc -o /data/sdd/zhaqiayi/salmon_try/fastqc/ -t 12 /data/raw_data/linpeng/MEPCE_RNA/468-sg*-?_?.fq.gz</span><br></pre></td></tr></table></figure>

<p>We use <strong>&#x3D;&#x3D;Wildcard (or Regular Expression)&#x3D;&#x3D;</strong> for batch scanning. *  represents any characters while ? stands for one character.</p>
<h4 id="Output-for-fastQC"><a href="#Output-for-fastQC" class="headerlink" title="Output for fastQC:"></a><strong>Output for fastQC:</strong></h4><p><img src="/./RNA-Seq-upstream-workflow/fastqc_output.png"></p>
<p>The results for sequencing are summarized in the HTML file. We use the reports to check whether the adapters have to be trimmed.</p>
<p>Luckily, with the help of <strong>multiQC</strong>, we don’t have to check all the reports. <strong>MultiQC</strong> could compile the fastqc reports into one file. This software can also be applied to other results. </p>
<h4 id="Run-multiQC"><a href="#Run-multiQC" class="headerlink" title="Run multiQC:"></a><strong>Run multiQC:</strong></h4><p><strong>Input:</strong> fastqc results</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">multiqc /data/sdd/zhaqiayi/salmon_try/fastqc/ -o /data/sdd/zhaqiayi/salmon_try/fastqc/report</span><br></pre></td></tr></table></figure>


<h4 id="Output-for-multiQC"><a href="#Output-for-multiQC" class="headerlink" title="Output for multiQC:"></a><strong>Output for multiQC:</strong></h4><p><img src="/./RNA-Seq-upstream-workflow/multiqc_output.png"></p>
<h4 id="Result-analysis"><a href="#Result-analysis" class="headerlink" title="Result analysis:"></a><strong>Result analysis:</strong></h4><p>The most important thing is to check <strong>whether the adapter of sequencing result has been trimmed.</strong> We could also fully understand the report content via the following<br><a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/20731723">link</a>.</p>
<h3 id="Trim-galore"><a href="#Trim-galore" class="headerlink" title="Trim_galore"></a>Trim_galore</h3><p>Trim_galore can <strong>automatically identify the adapter regions</strong>. So, we just input the file and obtain results without complex operations.</p>
<h4 id="Install-1"><a href="#Install-1" class="headerlink" title="Install:"></a><strong>Install:</strong></h4><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">conda install trim_galore -c bioconda</span><br></pre></td></tr></table></figure>


<h4 id="Run"><a href="#Run" class="headerlink" title="Run:"></a><strong>Run:</strong></h4><p><strong>Input:</strong> .fq.gz file from raw data</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#trim_galore</span></span><br><span class="line"><span class="comment">#paired-end input</span></span><br><span class="line"><span class="comment">#-o output dir</span></span><br><span class="line">trim_galore --gzip -j 12 --paired -o /data/sdd/zhaqiayi/fastqc1 NTm-2_S1_L001_R1_001.fastq.gz NTm-2_S1_L001_R2_001.fastq.gz</span><br></pre></td></tr></table></figure>

<p>After the adapter being trimmed, we <strong>run fastQC to verify</strong> it.</p>
<h3 id="Salmon"><a href="#Salmon" class="headerlink" title="Salmon"></a>Salmon</h3><h4 id="Introduction"><a href="#Introduction" class="headerlink" title="Introduction:"></a><strong>Introduction:</strong></h4><p>Salmon is capable of either mapping sequencing reads itself by using a fast and lightweight procedure called <strong>quasi-mapping</strong> or accepting precomputed read alignments in the form of a SAM or BAM file.<br><strong>Unlike pseudoalignment</strong>, Salmon’s lightweight mapping procedure tracks, by default, the position and orientation of all mapped fragments. This information is used in conjunction with the abundances from online inference to <strong>compute per-fragment conditional probabilities, rather than just transcript compatibility</strong> (Patro et al., 2017).  <a target="_blank" rel="noopener" href="https://sci-hub.ru/10.1038/nmeth.4197">Paper Link</a></p>
<h4 id="Install-2"><a href="#Install-2" class="headerlink" title="Install:"></a><strong>Install:</strong></h4><ol>
<li>Download the salmon-1.9.0_linux_x86_64 file for linux from <a target="_blank" rel="noopener" href="https://github.com/COMBINE-lab/salmon/releases">release page</a>.</li>
<li>Unzip file.</li>
</ol>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">tar xzvf salmon-1.9.0_linux_x86_64.tar.gz</span><br></pre></td></tr></table></figure>
<ol start="3">
<li>We could use it with <strong>absolute path</strong> now!</li>
</ol>
<h4 id="Run-1"><a href="#Run-1" class="headerlink" title="Run:"></a><strong>Run:</strong></h4><p>We are supposed to compile all the codes for Salmon into one shell document. Then we just run:</p>
<p><strong>Input:</strong> .fq.gz file with adapter trimmed</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#test.sh: file name</span></span><br><span class="line">bash test.sh</span><br></pre></td></tr></table></figure>

<p>The content of bash file is as follow:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#Define the variables: Usually the path</span></span><br><span class="line"><span class="comment">#Find: find out files with patterns </span></span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> /data/sdd/zhaqiayi/salmon_try</span><br><span class="line">list1=$(find /data/raw_data/linpeng/MEPCE_RNA/ -name <span class="string">&quot;*1.fq.gz&quot;</span>)</span><br><span class="line">list2=$(find /data/raw_data/linpeng/MEPCE_RNA/ -name <span class="string">&quot;*2.fq.gz&quot;</span>)</span><br><span class="line">list3=$(find /data/raw_data/linpeng/MEPCE_RNA/ -name <span class="string">&quot;*1.fq.gz&quot;</span> |sed <span class="string">&#x27;s#.*/##&#x27;</span>)</span><br><span class="line">ref_genome=/software/ReferenceData/Human/GRCh38_hg38.p13/Software_index/salmonV1.9_EnsemblV108</span><br></pre></td></tr></table></figure>
<p><strong>Notes:</strong></p>
<ol>
<li>Define variables: There is <strong>no blank space between ‘&#x3D;’</strong>.</li>
<li>Some parameters can be easily change for different projects by <strong>defining variables</strong>. </li>
<li>Use <strong>find</strong> command to search for files: learning <strong>wildcard (or regular expression)</strong> thoroughly also matters.</li>
<li>Don ‘t loss yourself in mess. :(</li>
</ol>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#transfer string to list</span></span><br><span class="line">list1_x=(<span class="variable">$list1</span>)</span><br><span class="line">list2_x=(<span class="variable">$list2</span>)</span><br><span class="line">list3_x=(<span class="variable">$list3</span>)</span><br><span class="line"></span><br><span class="line"><span class="comment">#length</span></span><br><span class="line">length=<span class="variable">$&#123;#list1_x[@]&#125;</span> </span><br><span class="line"></span><br><span class="line"><span class="comment">#for loop</span></span><br><span class="line"><span class="comment">#use absolute path for calling salmon program</span></span><br><span class="line"><span class="comment">#.* any character</span></span><br><span class="line"><span class="keyword">for</span> ((i=<span class="number">0</span>; i&lt;<span class="variable">$&#123;length&#125;</span>; i++));</span><br><span class="line"><span class="keyword">do</span></span><br><span class="line">   /data/sdd/zhaqiayi/download/salmon-1.9.0_linux_x86_64/bin/salmon quant -i <span class="variable">$ref_genome</span> -l A -1 <span class="variable">$&#123;list1_x[$i]&#125;</span>  -2 <span class="variable">$&#123;list1_x[$i]/%1.fq.gz/2.fq.gz&#125;</span> -o /data/sdd/zhaqiayi/salmon_try/<span class="variable">$&#123;list3_x[$i]/%.fq.gz/&#125;</span>/ -p 12</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>

<p><strong>Notes:</strong></p>
<ol>
<li>Use <strong>absolute path for calling salmon program</strong>, which doesn’t rely on environment. </li>
<li><strong>Pattern matching</strong> for a language.</li>
</ol>
<p><strong>Improved version:</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="built_in">cd</span> /data/sdd/zhaqiayi/salmon_try</span><br><span class="line">list1=$(find /data/raw_data/linpeng/MEPCE_RNA/ -name <span class="string">&quot;*1.fq.gz&quot;</span>)</span><br><span class="line"></span><br><span class="line">ref_genome=/software/ReferenceData/Human/GRCh38_hg38.p13/Software_index/salmonV1.9_EnsemblV108</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> file1 <span class="keyword">in</span> <span class="variable">$list1</span>;</span><br><span class="line"><span class="keyword">do</span>    </span><br><span class="line">    file2=<span class="variable">$&#123;file1/%1.fq.gz/2.fq.gz&#125;</span>;</span><br><span class="line">    <span class="comment"># basename: linux function for filename </span></span><br><span class="line">    fileName=`<span class="built_in">basename</span> <span class="variable">$file1</span>`; </span><br><span class="line">    <span class="comment"># sed: _.* -&gt; after _ to nothing</span></span><br><span class="line">    fileName=`<span class="built_in">echo</span> <span class="variable">$fileName</span> | sed <span class="string">&#x27;s/_.*//&#x27;</span>`;</span><br><span class="line">    output_dir=/data/sdd/zhaqiayi/salmon_try/<span class="variable">$fileName</span></span><br><span class="line">    /data/sdd/zhaqiayi/download/salmon-1.9.0_linux_x86_64/bin/salmon quant -i <span class="variable">$ref_genome</span> -l A -1 <span class="variable">$file1</span>  -2 <span class="variable">$file2</span> -o <span class="variable">$output_dir</span> -p 12</span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>











    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/RNA-Seq/" rel="tag"># RNA-Seq</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item"></div>
      <div class="post-nav-item">
    <a href="/2023/02/24/001-Linux%E5%AD%A6%E4%B9%A001-%E5%AE%89%E8%A3%85%E7%94%9F%E4%BF%A1%E8%BD%AF%E4%BB%B6%E7%9A%84%E9%82%A3%E4%BA%9B%E5%9D%91/" rel="next" title="001- Linux学习01-安装生信软件的那些坑">
      001- Linux学习01-安装生信软件的那些坑 <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#Overview"><span class="nav-number">1.</span> <span class="nav-text">Overview</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#FastQC-Multiqc"><span class="nav-number">2.</span> <span class="nav-text">FastQC (+ Multiqc)</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Install"><span class="nav-number">2.1.</span> <span class="nav-text">Install:</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Run-fastQC"><span class="nav-number">2.2.</span> <span class="nav-text">Run fastQC:</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Output-for-fastQC"><span class="nav-number">2.3.</span> <span class="nav-text">Output for fastQC:</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Run-multiQC"><span class="nav-number">2.4.</span> <span class="nav-text">Run multiQC:</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Output-for-multiQC"><span class="nav-number">2.5.</span> <span class="nav-text">Output for multiQC:</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Result-analysis"><span class="nav-number">2.6.</span> <span class="nav-text">Result analysis:</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Trim-galore"><span class="nav-number">3.</span> <span class="nav-text">Trim_galore</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Install-1"><span class="nav-number">3.1.</span> <span class="nav-text">Install:</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Run"><span class="nav-number">3.2.</span> <span class="nav-text">Run:</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Salmon"><span class="nav-number">4.</span> <span class="nav-text">Salmon</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#Introduction"><span class="nav-number">4.1.</span> <span class="nav-text">Introduction:</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Install-2"><span class="nav-number">4.2.</span> <span class="nav-text">Install:</span></a></li><li class="nav-item nav-level-4"><a class="nav-link" href="#Run-1"><span class="nav-number">4.3.</span> <span class="nav-text">Run:</span></a></li></ol></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Qiayi"
      src="http://img.name2012.com/uploads/allimg/2015-04/26-011155_74.jpg">
  <p class="site-author-name" itemprop="name">Qiayi</p>
  <div class="site-description" itemprop="description">An island where we record bioinformatics learning process.</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">9</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">7</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">6</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2023-02 – 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Qiayi</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
