<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 6.3.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">


<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"example.com","root":"/","scheme":"Muse","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":false,"show_result":false,"style":null},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":false,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}}};
  </script>

  <meta name="description" content="Overview pre-alignment QC: FastQC -&gt; Trim-galore -&gt; FastQC  alignment: Bowtie2&#x2F;BWA post-alignment QC: cross-correlation analysis call peaks: MACS2 QC: FRiP&#x2F;RiBL  It seems that we shoul">
<meta property="og:type" content="article">
<meta property="og:title" content="ChIP-seq upstream analysis">
<meta property="og:url" content="http://example.com/2023/03/19/ChIP-seq-upstream-analysis/index.html">
<meta property="og:site_name" content="BIOINFOISLAND">
<meta property="og:description" content="Overview pre-alignment QC: FastQC -&gt; Trim-galore -&gt; FastQC  alignment: Bowtie2&#x2F;BWA post-alignment QC: cross-correlation analysis call peaks: MACS2 QC: FRiP&#x2F;RiBL  It seems that we shoul">
<meta property="og:locale" content="en_US">
<meta property="article:published_time" content="2023-03-19T11:53:19.000Z">
<meta property="article:modified_time" content="2023-04-12T11:17:39.648Z">
<meta property="article:author" content="Qiayi">
<meta property="article:tag" content="ChIP-seq">
<meta name="twitter:card" content="summary">

<link rel="canonical" href="http://example.com/2023/03/19/ChIP-seq-upstream-analysis/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'en'
  };
</script>

  <title>ChIP-seq upstream analysis | BIOINFOISLAND</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="Toggle navigation bar">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">BIOINFOISLAND</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>Home</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>About</a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>categories</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>Tags</a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>Archives</a>

  </li>
  </ul>
</nav>




</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="en">
    <link itemprop="mainEntityOfPage" href="http://example.com/2023/03/19/ChIP-seq-upstream-analysis/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="http://img.name2012.com/uploads/allimg/2015-04/26-011155_74.jpg">
      <meta itemprop="name" content="Qiayi">
      <meta itemprop="description" content="An island where we record bioinformatics learning process.">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="BIOINFOISLAND">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          ChIP-seq upstream analysis
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">Posted on</span>

              <time title="Created: 2023-03-19 19:53:19" itemprop="dateCreated datePublished" datetime="2023-03-19T19:53:19+08:00">2023-03-19</time>
            </span>
              <span class="post-meta-item">
                <span class="post-meta-item-icon">
                  <i class="far fa-calendar-check"></i>
                </span>
                <span class="post-meta-item-text">Edited on</span>
                <time title="Modified: 2023-04-12 19:17:39" itemprop="dateModified" datetime="2023-04-12T19:17:39+08:00">2023-04-12</time>
              </span>
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-folder"></i>
              </span>
              <span class="post-meta-item-text">In</span>
                <span itemprop="about" itemscope itemtype="http://schema.org/Thing">
                  <a href="/categories/ChIP-seq-Workflow/" itemprop="url" rel="index"><span itemprop="name">ChIP-seq Workflow</span></a>
                </span>
            </span>

          

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h3 id="Overview"><a href="#Overview" class="headerlink" title="Overview"></a>Overview</h3><ul>
<li>pre-alignment QC: FastQC -&gt; Trim-galore -&gt; FastQC </li>
<li>alignment: Bowtie2&#x2F;BWA</li>
<li>post-alignment QC: cross-correlation analysis</li>
<li>call peaks: MACS2</li>
<li>QC: FRiP&#x2F;RiBL</li>
</ul>
<p>It seems that we should conduct quality control after each step.</p>
<span id="more"></span>

<h3 id="pre-alignment-QC-FastQC-amp-multiqc"><a href="#pre-alignment-QC-FastQC-amp-multiqc" class="headerlink" title="pre-alignment QC: FastQC &amp; multiqc"></a>pre-alignment QC: FastQC &amp; multiqc</h3><p>This part is similar to that in RNA-seq. We just run this as what we did before.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># fastQC</span></span><br><span class="line"><span class="comment"># -o output dir </span></span><br><span class="line"><span class="comment"># -t thread</span></span><br><span class="line"><span class="comment"># input dir is at the end of the command</span></span><br><span class="line"></span><br><span class="line">fastqc -o /data/sdd/zhaqiayi/chip/mar/00-qc -t 12 /data/sdd/guochen/20230102/20221231/*.fq.gz</span><br><span class="line">multiqc /data/sdd/zhaqiayi/chip/mar/00-qc -o /data/sdd/zhaqiayi/chip/mar/00-qc</span><br><span class="line"></span><br></pre></td></tr></table></figure>

<p>We find out that most of the adapters are not trimmed.</p>
<p><strong>output:</strong><br>Click here: <a target="_blank" rel="noopener" href="https://postimg.cc/9rw71Nkr">fastqc1.png</a> </p>
<h3 id="Trim-galore-amp-QC"><a href="#Trim-galore-amp-QC" class="headerlink" title="Trim-galore &amp; QC"></a>Trim-galore &amp; QC</h3><p>We can trim the adapters. Here we use <code>for loop</code> as before. </p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line">list1=$(find /data/sdd/guochen/20230102/20221231/ -name <span class="string">&quot;*1_001.fastq.gz&quot;</span>)</span><br><span class="line"></span><br><span class="line"><span class="keyword">for</span> file1 <span class="keyword">in</span> <span class="variable">$list1</span>;</span><br><span class="line"><span class="keyword">do</span>    </span><br><span class="line">    file2=<span class="variable">$&#123;file1/%1_001.fastq.gz/2_001.fastq.gz&#125;</span>;</span><br><span class="line">    <span class="comment"># basename: linux function for filename </span></span><br><span class="line">    fileName=`<span class="built_in">basename</span> <span class="variable">$file1</span>`; </span><br><span class="line">    <span class="comment"># sed: _.* -&gt; after _ to nothing</span></span><br><span class="line">    fileName=`<span class="built_in">echo</span> <span class="variable">$fileName</span> | sed <span class="string">&#x27;s/_.*//&#x27;</span>`;</span><br><span class="line">    output_dir=/data/sdd/zhaqiayi/chip/mar/01-trim/<span class="variable">$fileName</span></span><br><span class="line">    trim_galore --gzip -j 12 --paired -o <span class="variable">$output_dir</span> <span class="variable">$file1</span> <span class="variable">$file2</span> </span><br><span class="line"><span class="keyword">done</span></span><br></pre></td></tr></table></figure>

<p><strong>output:</strong><br><a target="_blank" rel="noopener" href="https://postimg.cc/qNLDMRGf">trim.png</a></p>
<p>Then, run fastqc again to check whether the adapters are successfully trimmed.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">fastqc -o /data/sdd/zhaqiayi/chip/mar/03-qc -t 12 /data/sdd/zhaqiayi/chip/mar/02-trim*.fq.gz</span><br><span class="line">multiqc /data/sdd/zhaqiayi/chip/mar/03-qc -o /data/sdd/zhaqiayi/chip/mar/03-qc</span><br></pre></td></tr></table></figure>

<p>In this step we could also check the <strong>duplication level</strong>, or to say, <strong>library complexity</strong> from fastQC report.</p>
<p>Here, we disappointed find out that the duplication level is really high（&gt;50%）, which means there are a lot of <strong>PCR bias</strong> in this data.</p>
<p><strong>fastQC result：</strong><br>Click here: <a target="_blank" rel="noopener" href="https://postimg.cc/BPvPCgVp">duplication-level.png</a><br>Please refer to this link for more details: <a target="_blank" rel="noopener" href="https://zhuanlan.zhihu.com/p/71893930">high duplication</a></p>
<h3 id="Alignment-Bowtie2"><a href="#Alignment-Bowtie2" class="headerlink" title="Alignment: Bowtie2"></a>Alignment: Bowtie2</h3><p>After being trimmed the adapters, the reads of the sequence can be mapped to the reference genome. <code>bowtie2</code> and <code>BWA</code> can both be used for this purpose. <strong><code>bowtie2</code> is much more faster since the alignment is not that strict as <code>BWA</code>.</strong> However, we could also change the extent of sensitivity while aligning the reads with <code>bowtie2</code>.</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># run bowtie2</span></span><br><span class="line"><span class="comment"># 这里索引的路径记得加上索引前缀</span></span><br><span class="line"><span class="comment"># 可以用for循环来批量处理数据</span></span><br><span class="line"></span><br><span class="line">bowtie2 -p 12 -x /software/ReferenceData/Human/GRCh38_hg38.p13/Software_index/bowtie2.V2.4.5_Ensembl.V108/GRCh38 -1 /data/sdd/guochen/20220626_human_BRD9_ChIP_newtest/20220622/M4_100000_plus_5/03trim_galore_out/M4_100000_plus_5_S92_L003_R1_001_val_1.fq.gz -2 /data/sdd/guochen/20220626_human_BRD9_ChIP_newtest/20220622/M4_100000_plus_5/03trim_galore_out/M4_100000_plus_5_S92_L003_R2_001_val_2.fq.gz -S /data/sdd/zhaqiayi/chip/bowtie2.sam</span><br></pre></td></tr></table></figure>

<p>After aligning the reads to genome, there is <strong>screen output</strong> where we could find out the alignment rate and many other results of this process. There is also <strong>another way to check the alignment quality</strong>: </p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># check the num of total reads and successfully mapped reads</span></span><br><span class="line">samtools flagstat bowtie2.sam &gt;bowtie2.flagstat</span><br><span class="line"><span class="built_in">cat</span> bowtie2.flagstat</span><br></pre></td></tr></table></figure>

<p>The details of comparison these two outputs can be seen here: (click)[]</p>
<h4 id="remove-duplications"><a href="#remove-duplications" class="headerlink" title="remove duplications:"></a>remove duplications:</h4><p>Since high duplication rate of the data, we should remove duplications of the reads before calling peaks.<br>Here are two ways of removing duplications:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Method 1: samtools</span></span><br><span class="line">samtools rmdup <span class="variable">$file1</span> /data/sdd/zhaqiayi/chip/mar/03-bowtie2/<span class="variable">$fileName</span>.rmdup.bam </span><br><span class="line"></span><br><span class="line"><span class="comment"># Method 2: sambamba </span></span><br><span class="line">sambamba view -h -t 12 -F <span class="string">&quot;[XS] == null and not unmapped and not duplicate&quot;</span> -f bam <span class="variable">$file1</span> -o <span class="variable">$fileName</span>.rmdup1.bam</span><br><span class="line"></span><br><span class="line"><span class="comment"># Method3: GATK picard</span></span><br><span class="line">java -jar picard.jar MarkDuplicates \</span><br><span class="line">      I=<span class="variable">$file1</span>.bam \</span><br><span class="line">      O=/data/sdd/zhaqiayi/chip/MEPCE/02-bowtie2/<span class="variable">$file1</span>.bam \</span><br><span class="line">      M=<span class="variable">$file1</span>.marked_dup_metrics.txt \</span><br><span class="line">      REMOVE_DUPLICATES=<span class="literal">true</span></span><br></pre></td></tr></table></figure>

<p><strong><code>samtools</code> is so slow that people design another tool called <code>sambamba</code> to speed up the process.</strong> The argument of these 2 tools are similar.<br><strong>It is recommended that <code>GATK picard</code> should be used in most of the projects.</strong></p>
<p>I think we could quickly check the alignment rate with flagstat and the duplication rate with fastqc. </p>
<h3 id="Post-alignment-QC-cross-correlation-analysis"><a href="#Post-alignment-QC-cross-correlation-analysis" class="headerlink" title="Post-alignment QC: cross-correlation analysis"></a>Post-alignment QC: cross-correlation analysis</h3><p>The cross correlation method is to find out <strong>whether the switch size is reasonable compared to the phantom peak.</strong> If the quality of experiment is ok, we can <strong>use the switch size for peak calling modeling.</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">Rscript run_spp.R -c=/data/sdd/zhaqiayi/chip/mar/sambamba-rmdup/B303h.sam.sorted.bam.rmdup1.bam -savp=/data/sdd/zhaqiayi/chip/mar/03-bowtie2/cc/B303h_xcor1.pdf -s=0:5:500</span><br></pre></td></tr></table></figure>

<h3 id="Call-peaks-MACS2"><a href="#Call-peaks-MACS2" class="headerlink" title="Call peaks: MACS2"></a>Call peaks: MACS2</h3><p>This is the H3K27Ac histone modification and paired ended case. I conducted the following steps:</p>
<ol>
<li><strong>Most of the histone modification cases are not suitable for MACS2 modeling.</strong> We utilizes cc analysis to determine the switch size and use <code>--nomodel</code> argument to modeling by ourselves.</li>
<li>Determine whether it is a <strong>broad peak or narrow peak</strong>. We could find papers for reference.</li>
</ol>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">callpeak -t /data/sdd/zhaqiayi/chip/mar/sambamba-rmdup/B293h.sam.sorted.bam.rmdup1.bam -f BAMPE -g hs -n B293h -q 0.05 --outdir /data/sdd/zhaqiayi/chip/mar/04-macs/no-ctrl/ --nomodel --extsize 175</span><br></pre></td></tr></table></figure>

<p>There are also many other tools for peak calling such as SPP, MACS3 for ATAC-seq, SEACR specific for cut &amp; run date.</p>
<h3 id="QC-FRiP"><a href="#QC-FRiP" class="headerlink" title="QC: FRiP"></a>QC: FRiP</h3><p>After peak calling, we can use PRiP (Fraction Reads in Peaks) to <strong>check how many reads are in the peaks. The number should no less than 20%. (not really, ATAC may call for &gt;2%)</strong></p>
<p>Here are 2 methods which shares the same results:</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Method1: featureCounts</span></span><br><span class="line">awk <span class="string">&#x27;BEGIN&#123;FS=OFS=&quot;\t&quot;; print &quot;GeneID\tChr\tStart\tEnd\tStrand&quot;&#125;&#123;print $4, $1, $2+1, $3, &quot;.&quot;&#125;&#x27;</span> /data/sdd/zhaqiayi/chip/mar/04-macs/broad-1e-5/NTmock.sam.sorted.bam.rmdup1.bam_peaks.broadPeak &gt; /data/sdd/zhaqiayi/chip/mar/04-macs/broad-1e-5/NTmock_peaks.saf</span><br><span class="line">featureCounts -p -a /data/sdd/zhaqiayi/chip/mar/04-macs/broad-1e-5/NTmock_peaks.saf -F SAF -o NTmock_FRiP.txt /data/sdd/zhaqiayi/chip/mar/sambamba-rmdup/NTmock.sam.sorted.bam.rmdup1.bam</span><br><span class="line"></span><br><span class="line"><span class="comment"># Method2:</span></span><br><span class="line"></span><br><span class="line"><span class="comment">#total reads</span></span><br><span class="line">samtools view -c /data/sdd/zhaqiayi/chip/mar/sambamba-rmdup/NT3h.sam.sorted.bam.rmdup1.bam</span><br><span class="line"></span><br><span class="line"><span class="comment">#reads in peaks</span></span><br><span class="line">bedtools <span class="built_in">sort</span> -i NT3h_peaks.narrowPeak \</span><br><span class="line">  | bedtools merge -i stdin | bedtools intersect -u -nonamecheck \</span><br><span class="line">  -a /data/sdd/zhaqiayi/chip/mar/sambamba-rmdup/NT3h.sam.sorted.bam.rmdup1.bam -b stdin -ubam | samtools view -c</span><br><span class="line"></span><br><span class="line"><span class="comment">#(reads in peaks/total reads)</span></span><br></pre></td></tr></table></figure>

<h3 id="Visualization"><a href="#Visualization" class="headerlink" title="Visualization"></a>Visualization</h3><p>The first step of visualization is to <strong>convert <code>bam</code> file into <code>bw</code> file</strong>. The bigWig format is an indexed binary format useful for dense, continuous data that will be displayed in a genome browser as a graph&#x2F;track, but also is used as input for some of the visualization commands we will be running in <code>deepTools</code>.<br><code>deepTools</code> is a suite of Python tools developed for the efficient analysis of high-throughput sequencing data. <code>deepTools</code> has a wide variety of commands.  </p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># make index of bam files</span></span><br><span class="line">samtools index <span class="variable">$file</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># convert `bam` file into `bw` file</span></span><br><span class="line">bamCoverage -b /data/sdd/zhaqiayi/chip/mar/sambamba-rmdup/B29mock.sam.sorted.bam.rmdup1.bam \</span><br><span class="line">-o /data/sdd/zhaqiayi/chip/mar/05-visualization/B29mock.bw \</span><br><span class="line">--binSize 20 \</span><br><span class="line">--normalizeUsing BPM \</span><br><span class="line">--extendReads 175 \</span><br><span class="line">-p 12</span><br></pre></td></tr></table></figure>

<h3 id="TSS-plot"><a href="#TSS-plot" class="headerlink" title="TSS plot"></a>TSS plot</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">plotProfile -m /data/sdd/zhaqiayi/chip/mar/05-visualization/3h.gz \</span><br><span class="line">-out /data/sdd/zhaqiayi/chip/mar/05-visualization/TSS_3h.png \</span><br><span class="line">--perGroup \</span><br><span class="line">--colors green purple blue\</span><br><span class="line">--refPointLabel <span class="string">&quot;TSS&quot;</span> \</span><br><span class="line">-T <span class="string">&quot;read density&quot;</span> \</span><br><span class="line">-z <span class="string">&quot;&quot;</span></span><br></pre></td></tr></table></figure>

<h3 id="IGV-visualization"><a href="#IGV-visualization" class="headerlink" title="IGV visualization"></a>IGV visualization</h3>
    </div>

    
    
    

      <footer class="post-footer">
          <div class="post-tags">
              <a href="/tags/ChIP-seq/" rel="tag"># ChIP-seq</a>
          </div>

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2023/03/19/012-samtools%20flagstat%E5%91%BD%E4%BB%A4alignment%20rate%20%E7%BB%93%E6%9E%9C%E8%A7%A3%E8%AF%BB/" rel="prev" title="012-samtools flagstat命令alignment rate 结果解读">
      <i class="fa fa-chevron-left"></i> 012-samtools flagstat命令alignment rate 结果解读
    </a></div>
      <div class="post-nav-item"></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          Table of Contents
        </li>
        <li class="sidebar-nav-overview">
          Overview
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#Overview"><span class="nav-number">1.</span> <span class="nav-text">Overview</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#pre-alignment-QC-FastQC-amp-multiqc"><span class="nav-number">2.</span> <span class="nav-text">pre-alignment QC: FastQC &amp; multiqc</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Trim-galore-amp-QC"><span class="nav-number">3.</span> <span class="nav-text">Trim-galore &amp; QC</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Alignment-Bowtie2"><span class="nav-number">4.</span> <span class="nav-text">Alignment: Bowtie2</span></a><ol class="nav-child"><li class="nav-item nav-level-4"><a class="nav-link" href="#remove-duplications"><span class="nav-number">4.1.</span> <span class="nav-text">remove duplications:</span></a></li></ol></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Post-alignment-QC-cross-correlation-analysis"><span class="nav-number">5.</span> <span class="nav-text">Post-alignment QC: cross-correlation analysis</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Call-peaks-MACS2"><span class="nav-number">6.</span> <span class="nav-text">Call peaks: MACS2</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#QC-FRiP"><span class="nav-number">7.</span> <span class="nav-text">QC: FRiP</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Visualization"><span class="nav-number">8.</span> <span class="nav-text">Visualization</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#TSS-plot"><span class="nav-number">9.</span> <span class="nav-text">TSS plot</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#IGV-visualization"><span class="nav-number">10.</span> <span class="nav-text">IGV visualization</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="Qiayi"
      src="http://img.name2012.com/uploads/allimg/2015-04/26-011155_74.jpg">
  <p class="site-author-name" itemprop="name">Qiayi</p>
  <div class="site-description" itemprop="description">An island where we record bioinformatics learning process.</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">16</span>
          <span class="site-state-item-name">posts</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">7</span>
        <span class="site-state-item-name">categories</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">8</span>
        <span class="site-state-item-name">tags</span></a>
      </div>
  </nav>
</div>



      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2023-02 – 
  <span itemprop="copyrightYear">2023</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Qiayi</span>
</div>
  <div class="powered-by">Powered by <a href="https://hexo.io/" class="theme-link" rel="noopener" target="_blank">Hexo</a> & <a href="https://muse.theme-next.org/" class="theme-link" rel="noopener" target="_blank">NexT.Muse</a>
  </div>

        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/muse.js"></script>


<script src="/js/next-boot.js"></script>




  















  

  

</body>
</html>
